syncstorage/storage/sql/dbconnect.py

```diff
@@ -144,21 +159,85 @@ def _get_bso_columns(table_name):
 
 bso = Table("bso", metadata, *_get_bso_columns("bso"))
 
+# Table mapping (user_id, collection_id) => batch IDs
+
+batch_uploads = Table(
+    "batch_uploads",
+    metadata,
+    Column("batch", BigInteger, primary_key=True, nullable=False),
+    Column("userid", Integer, nullable=False, autoincrement=False),
+    Column("collection", Integer, nullable=False, autoincrement=False),
+    Column("open", Boolean, default=True)
+)
+
+# Column definitions for batch upload item table(s)
+#
+# These columns hold the batched uploads until either the client declares a
+# commit or the TTL expires.
+
+
+def _get_batch_item_columns(table_name):
+    identifiers = (
+        Column("batch", BigInteger, primary_key=True, nullable=False,
+               autoincrement=False),
+        Column("item", String(64), primary_key=True, nullable=False,
+               autoincrement=False)
+    )
+    common = _common_columns(MAX_TTL)
+    indexes = (
+        # Declare indexes.
+        # We need to include the tablename in the index name due to sharding,
+        # because index names in sqlite are global, not per-table.
+        # Index on "ttl" for easy pruning of expired items.
+        Index("%s_ttl_idx" % (table_name,), "ttl"),
```

Comment:
I'm not convinced we'll ever want to prune expired items from this table, but rather just prune everything that's been committed.  In fact, do we need anything other than the primary key index on this table?